Tests/js/apps/assets/ez-platformuiapp-tests.js

```diff
@@ -1561,6 +1561,63 @@ YUI.add('ez-platformuiapp-tests', function (Y) {
         },
     });
 
+    configRouteTest = new Y.Test.Case({
+        name: "eZ Platform UI App reverse routing tests",
+
+        setUp: function () {
+
+            this.root = '/this/is/the/root/';
+            this.app = new Y.eZ.PlatformUIApp({
+                root: this.root,
+                container: '.app',
+                viewContainer: '.view-container',
+                routeConfig: {
+                    "loginForm": {
+                        "fieldsViews": {
+                            "ezthing": 'Something'
+                        }
+                    },
+                    "doesNotMatchAnything": {
+                        "fieldsViews": {
+                            "ezthing": 'Anything'
+                        }
+                    },
+                }
+            });
+        },
+
+        tearDown: function () {
+            this.app.destroy();
+        },
+
+        "Should enrich route with config": function () {
+            var that = this;
+
+            Y.Array.each(this.app.get('routes'), function (value) {
+                if (value.name == 'loginForm') {
+                    Y.Assert.areSame(value.config, that.app.get('routeConfig').loginForm, 'Route loginForm should be enrich with config');
+                } else {
+                    Y.Assert.isUndefined(value.config, 'Route should not have been enrich if routeConfig does NOT match');
+                }
+            });
+        },
+
+        "Should Not enrich route with config if routeConfig does NOT match": function () {
+            var that = this;
+
+            Y.Array.each(this.app.get('routes'), function (value) {
+                if (value.name == 'loginForm') {
```

Comment:
in this test, you should actually that the `value.config` is never the same as `that.app.get('routeConfig').doesNotMatchAnything` (on the loginForm route and the others)